Update README.md to improve new deployment instructions (#422)
stackhpc/ansible-slurm-appliance: docs/operations.md

All subsequent sections assume that:
- Commands are run from the repository root, unless otherwise indicated by a `cd` command.
- An Ansible vault secret is configured.
- The correct private key is available to Ansible.
- Appropriate OpenStack credentials are available.
- Any non-appliance controlled infrastructure is avaialble (e.g. networks, volumes, etc.).
- `$ENV` is your current, activated environment, as defined by e.g. `environments/production/`.
- `$SITE_ENV` is the base site-specific environment, as defined by e.g. `environments/mysite/`.
- A string `some/path/to/file.yml:myvar` defines a path relative to the repository root and an Ansible variable in that file.
- Configuration is generally common to all environments at a site, i.e. is made in `environments/$SITE_ENV` not `environments/$ENV`.

Review any [site-specific documentation](site/README.md) for more details on the above.

# Deploying a Cluster

This follows the same process as defined in the main [README.md](../README.md) for the default configuration.

Note that tags as defined in the various sub-playbooks defined in `ansible/` may be used to only run part of the tasks in `site.yml`.

> # SSH to Cluster Nodes
> This depends on how the cluster is accessed.

Reviewer: New line missing after various headings
```suggestion
# SSH to Cluster Nodes

This depends on how the cluster is accessed.
```

The script `dev/ansible-ssh` may generally be used to connect to a host specified by a `inventory_hostname` using the same connection details as Ansible. If this does not work:
- Instance IPs are normally defined in `ansible_host` variables in an inventory file `environments/$ENV/inventory/hosts{,.yml}`.
- The ssh user is defined by `ansible_user`, default is `rocky`. This may be overriden in your environment.
- If a jump host is required the user and address may be defined in the above inventory file.

# Modifying general Slurm.conf parameters

> Parameters for [slurm.conf](https://slurm.schedmd.com/slurm.conf.html) can be added to an `openhpc_config_extra` mapping in `environments/$SITE_ENV/inventory/group_vars/all/openhpc.yml`.

Reviewer: We seem to alternate between `$ENV` and `$SITE_ENV` in various places, is this intentional?
```suggestion
Parameters for [slurm.conf](https://slurm.schedmd.com/slurm.conf.html) can be added to an `openhpc_config_extra` mapping in `environments/$ENV/inventory/group_vars/all/openhpc.yml`.
```

Author: Yes, from first section on this page:

> - `$ENV` is your current, activated environment, as defined by e.g. `environments/production/`.
> - `$SITE_ENV` is the base site-specific environment, as defined by e.g. `environments/mysite/`.

generally we want people to put config in the site env.

# Running the MPI Test Suite

See [ansible/roles/hpctests/README.md](ansible/roles/hpctests/README.md) for a description of these. They can be run using

    ansible-playbook ansible/adhoc/hpctests.yml

Note that:
- The above role provides variables to select specific partitions, nodes and interfaces which may be required. If not set in inventory, these can be passed as extravars:

        ansible-playbook ansible/adhoc/hpctests.yml -e hpctests_myvar=foo
- The HPL-based test is only resonably optimised on Intel processors due the libaries and default parallelisation scheme used. For AMD processors it is recommended this
is skipped using:

        ansible-playbook ansible/adhoc/hpctests.yml --skip-tags hpl-solo.

Review any [site-specific documentation](site/README.md) for more details.

# Running CUDA Tests
This uses the [cuda-samples](https://github.com/NVIDIA/cuda-samples/) utilities "deviceQuery" and "bandwidthTest" to test GPU functionality. It automatically runs on any
host in the `cuda` inventory group:

    ansible-playbook ansible/adhoc/cudatests.yml

**NB:** This test is not launched through Slurm, so confirm nodes are free/out of service or use `--limit` appropriately.

> # Ad-hoc Commands

Reviewer: I think it is worth mentioning the ability to run arbitrary shell commands against ansible groups here i.e. `ansible <group/host> -m shell -a "<shell-command>" (optional: --become)`. I know it's just standard ansible, but at least I wasn't aware of how useful it was until relatively recently.

```suggestion
# Ad-hoc Commands and Playbooks
```

> 5. "Utility" playbooks for managing a running appliance are contained in `ansible/adhoc` - run these by activating the environment and using:

Reviewer: ```suggestion
A set of utility playbooks for managing a running appliance are provided in `ansible/adhoc` - run these by activating the desired environment and using:
```
